src/shared/watchdog.c

```diff
@@ -482,6 +482,23 @@ int watchdog_ping(void) {
         return watchdog_ping_now();
 }
 
+void watchdog_report_missing(void) {
+        /* If we failed to open the watchdog because the device doesn't exist, report why. If we successfully
+         * opened a device, or opening failed for a different reason, we logged then. But ENOENT failure may
+         * be transient, for example because modules being loaded or the hardware is being detected. This
+         * function can be called to log after things have settled down.
+         *
+         * If a device was specified explicitly, raise level. */
+
+        if (watchdog_fd == -ENOENT)
+                log_full_errno(watchdog_device ? LOG_WARNING : LOG_NOTICE,
+                               watchdog_fd,
+                               "Failed to open %swatchdog device%s%s: %m",
```

Comment:
i'd clarify in the message when/why we are logging this, i.e. include "even after initial transaction completed"

maybe even make this a structured log message...